### ISSUE TITLE
[BUG] Set-PnPListItem not working on lists with more than 12 lookup columns

### ISSUE THREAD

**Issue by ohaak2** [pnp/powershell#4311]
### Expected behavior 
Using Set-PnPListItem to update some values of a list item in a list having more than 12 lookup columns shouldn't run into this lookup column threshold issue as long as we are not trying to set more than 12 lookup columns in that operation.

### Actual behavior 
We will get the 

### Steps to reproduce behavior
Please include complete script or code samples in-line or linked from [gists](https://gist.github.com/)

### What is the version of the Cmdlet module you are running?
(you can retrieve this by executing ```Get-Module -Name "PnP.PowerShell" -ListAvailable```)

### Which operating system/environment are you running PnP PowerShell on?
- [ ] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by ohaak2** [comment]
I've checked the code to clarify why PnP.PowerShell is obviouslty trying to query all list columns while I'm just trying to update e.g just one field.

Found the following code in class ListItemHelper:

![image](https://github.com/user-attachments/assets/e2766d84-1057-4fa0-bd55-40e11ecb318e)

I think it should be possible to query just the fields that are acutally rerference in the Cmdlets "value" Parameter. That way it should be easy to avoid that threshold issue.

I've verified my theory using the Set-PnPTaxonomyFieldValue Cmdlet on that same list - with no issues at all. Here the code (in Microsoft.SharePointClient.TaxonomyExtensions)  is just loading the field in question and everything works fine:

![image](https://github.com/user-attachments/assets/8832c6c5-e9d1-4822-88ee-7674c513817f)

**Comment by ohaak2** [comment]
By the way - this is the error message we're currently facing:
_The query cannot be completed because the number of lookup columns it contains exceeds the lookup column threshold._

I'm even getting this error, if I'm just trying to change the item's content type using Set-PnPListItem. No additional values involved. And all this just because Set-FieldValues is trying to load all fields involved (for whatever reason).

**Comment by Tanddant** [comment]
As a workaround, what happens if you load the item and updated it manually?

Something like this (don't have PowerShell handy right now, but you should be able to get it to work)
```PowerShell
$li = Get-PnPListItem -Identity 1
$li["Fieldname"] = "New Value"
$li.update();
Invoke-PnPQuery
```

**Comment by ohaak2** [comment]
@Tanddant  Yes, indeed. That's working for me. I've already tested a fix for the code behind Set-PnPListItem. The trick is to avoid the query for all fields as mentioned in my post above:

![image](https://github.com/user-attachments/assets/6448c715-1390-4ce5-8116-552eed08d5b3)

I'm avoiding to load all fields by query. The only fields the code now loads are the ones involved, if you provide any value updates. 

My fix so far just involved the single item update - not the batch operation. The latter one is unfortunately far more tricky.

**Comment by gautamdsheth** [comment]
This seems to coming from `get-pnplistitem` and maybe using some query or something. We fixed this by adding a better warning for that scenario.

Also, fixed the `set-pnplistitem` to only fetch the columns that we need to set. This reduced the network traffic considerably from 10 kb to 0.5 kb as we just fetch the needed columns instead of all columns (130-ish), should help you know. While we cant fix it if its > 12 columns in get , but the setting part should work with the latest nightly builds ! 

Thanks for reporting this issue @ohaak2 and @Tanddant for checking !!
